assets/tests/TaskSchemaValidationService.js

```diff
@@ -0,0 +1,262 @@
+        var buggyOutputTests = task.getBuggyOutputTests();
+        for (var i = 0; i < buggyOutputTests.length; i++) {
+          var messages = buggyOutputTests[i].getMessages();
+          if (!messages) {
+            return false;
+          }
+          if (!Array.isArray(messages)) {
+            return false;
+          }
+          for (var j = 0; j < messages.length; j++) {
+            var message = messages[j];
+            if (!(typeof message === 'string' || message instanceof String)) {
+              return false;
+            }
+          }
+        }
+        return true;
+      },
+      verifyPerformanceTestsAreArray: function(task) {
+        var performanceTests = task.getPerformanceTests();
+        return Array.isArray(performanceTests);
+      },
+      verifyPerformanceTestsHaveInputDataAtomString: function(
+        task) {
+        var performanceTests = task.getPerformanceTests();
+        for (var i = 0; i < performanceTests.length; i++) {
+          var test = performanceTests[i];
+          var inputDataAtom = test.getInputDataAtom();
+          if (!inputDataAtom) {
+            return false;
+          }
+          if (!(typeof inputDataAtom === 'string' ||
+            inputDataAtom instanceof String)) {
+            return false;
+          }
+        }
+        return true;
+      },
+      verifyPerformanceTestsHaveNonEmptyTransformationFunctionName: function(
+        task, auxiliaryCode) {
+        var performanceTests = task.getPerformanceTests();
+        for (var i = 0; i < performanceTests.length; i++) {
+          var test = performanceTests[i];
+          var transformationFunctionName = test.getTransformationFunctionName();
+          if (!transformationFunctionName) {
+            return false;
+          }
+          if (!(typeof transformationFunctionName === 'string' ||
+            transformationFunctionName instanceof String)) {
+            return false;
+          }
+          var auxiliaryCodeClassName = 'AuxiliaryCode.';
+          if (transformationFunctionName.startsWith(auxiliaryCodeClassName)) {
+            var functionName = transformationFunctionName.substring(
+              auxiliaryCodeClassName.length);
+            return auxiliaryCode.indexOf(functionName) !== -1;
+          }
+          var systemCodeClassName = 'System.';
+          if (transformationFunctionName.startsWith(systemCodeClassName)) {
+            // TODO(eyurko): Make system code accessible by testing framework.
+            return true;
+          }
+          return false;
+        }
+        return true;
+      },
+      verifyPerformanceTestsHaveLinearExpectedPerformance: function(
+        task) {
+        var performanceTests = task.getPerformanceTests();
+        for (var i = 0; i < performanceTests.length; i++) {
+          var test = performanceTests[i];
+          // TODO(eyurko): Update this once we support nonlinear runtimes.
+          if (test.getExpectedPerformance() !== 'linear') {
```

Comment:
Maybe have a constant in app.js somewhere saying `ALLOWED_RUNTIMES = ['linear', ...]`? Then import that here and (perhaps) in the code under test?